include/capstone/capstone.h

```diff
@@ -315,6 +315,7 @@ typedef struct cs_detail {
 		cs_wasm wasm;	///< Web Assembly architecture
 		cs_mos65xx mos65xx;	///< MOS65XX architecture (including MOS6502)
 	};
+	cs_arch arch; ///Used architecture (for architecture-specific free)
```

Comment:
this does not look good, but sounds like we cannot avoid adding this field? 

another solution is to have a new csh argument for cs_free()

Reply:
Indeed.
As a recap, I see three solutions :
* this patch (but more memory consumption)
* changing `cs_free` prototype (but changing interface is painful for people using this project)
* changing how WASM handles this "switch" jump table (but I am not sure if this relevant for WASM)

Which is the best ?

Reply:
yes we try best to avoid breaking compatibility, but in this case i think that is still better than having the extra arch field in cs_insn. finally, it was a mistake not to have csh arg in every API. we may afford changing API in the version 5, which will have major update in all archs.

if there is upper limit for brtable, we can avoid this mess by having fixed code array for operands. lets see what @spikel has to say when he is back.